### PR TITLE
Simpler version (without state & componentWillReceiveProps)

### LINKED ISSUE #9: Drop componentWillReceiveProps in favor of the more future-proof getDerivedStateFromProps
As outlined in https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#open-source-project-maintainers, it is recommended to use the new (react-async compatible) getDerivedStateFromProps rather than componentWillReceiveProps.

Libraries can use the react-lifecycles-compat polyfill to stay compatible with react < 16.3

### PR DESCRIPTION
**Pull request by yched**
Fixes #9 and #11

### CODE REVIEW

**Review by oliviertassinari — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -3,33 +3,19 @@ import debounce from 'lodash/debounce';
...
-            this.shouldRender = true;
-            this.setState(props);
-        }, ...debounceArgs);
-
-        shouldComponentUpdate() {
-            return this.shouldRender;
+            this.updateDebounced.cancel();
         }
-
+
```
Comment (amio) [inline src/index.js:16]: Extra spaces?

**Comment by himynameisdave** [comment]
Thanks for fixing these issues! I'd like to use this component in production but will need this to be merged & published first, any idea when we can make that happen?

Thanks again for this component, it's solid for debouncing updates to `react-apollo`'s `Query` component 👌👌👌 

Ping @podefr

File: src/index.js (hunk 1 above)
Comment (podefr) [reply src/index.js:16]: I'll clean this up :)

**Comment by podefr** [comment]
@yched I don't know how to thank you, this is amazing! It's a way simpler implementation than the original one and it also seems a lot cleaner. I'll merge this in and will publish a new version to npm.

**Comment by podefr** [comment]
@himynameisdave thanks for using react-debounce-render! Please try version 5.0.0 with Yves' changes and let me know if you have any issue :) thanks!

**Pull request merged**